valhalla/thor/pathinfo.h

```diff
@@ -12,33 +12,34 @@ namespace thor {
  * Simple(ish) structure to pass path information from PathAlgorithm
  * to TripLegBuilder
  */
+// TODO: just use the sif::Cost object for the pairs of floats below
 struct PathInfo {
-  sif::TravelMode mode; // Travel mode along this edge
-  float elapsed_time; // Elapsed time in seconds at the end of the edge including any turn cost at the
-                      // start of the edge
-  uint32_t trip_id;   // Trip Id (0 if not a transit edge).
-  baldr::GraphId edgeid; // Directed edge Id
-  float elapsed_cost;    // Cost to the end of the edge in cost units
-  int restriction_index; // Record which restriction
-  float turn_cost; // Turn cost in seconds at the beginning of the edge, only in map matching for now
+  sif::TravelMode mode;   // Travel mode along this edge
+  sif::Cost elapsed_cost; // Elapsed cost at the end of the edge including any turn cost at the start
+                          // of the edge
+  uint32_t trip_id;       // Trip Id (0 if not a transit edge).
+  baldr::GraphId edgeid;  // Directed edge Id
+  int restriction_index;  // Record which restrictionn
+  sif::Cost transition_cost; // Turn cost at the beginning of the edge
```

Comment:
here we store actual cost objects now. previously we had the floats broken out and we omitted the transitions cost and just had the seconds. now we store all the cost we need as cost objects